EmployeeManagementSystem/src/main/java/Option2.java

```diff
@@ -0,0 +1,187 @@
+        this.last_name = last_name;
+    }
+
+    @Override
+    public int compareTo(Employee o) {
+        if (o.getLastName().equals(this.last_name))
+            return 0;
+        else
+            return -1;
+    }
+}
+
+class MiddleNumberComparable implements Comparable<Employee> {
+    private String middle_number;
+    public MiddleNumberComparable(String middle_number) {
+        this.middle_number = middle_number;
+    }
+
+    @Override
+    public int compareTo(Employee o) {
+        if (o.getMiddleNumber().equals(this.middle_number))
+            return 0;
+        else
+            return -1;
+    }
+}
+
+class LastNumberComparable implements Comparable<Employee> {
+    private String last_number;
+    public LastNumberComparable(String last_number) {
+        this.last_number = last_number;
+    }
+
+    @Override
+    public int compareTo(Employee o) {
+        if (o.getLastNumber().equals(this.last_number))
+            return 0;
+        else
+            return -1;
+    }
+}
+
+class BirthYearComparable implements Comparable<Employee> {
+    private String birth_year;
+    public BirthYearComparable(String birth_year) {
+        this.birth_year = birth_year;
+    }
+
+    @Override
+    public int compareTo(Employee o) {
+        if (o.getBirthYear().equals(this.birth_year))
+            return 0;
+        else
+            return -1;
+    }
+}
+
+class BirthMonthComparable implements Comparable<Employee> {
+    private String birth_month;
+    public BirthMonthComparable(String birth_month) {
+        this.birth_month = birth_month;
+    }
+
+    @Override
+    public int compareTo(Employee o) {
+        if (o.getBirthMonth().equals(this.birth_month))
+            return 0;
+        else
+            return -1;
+    }
+}
+
+class BirthDayComparable implements Comparable<Employee> {
+    private String birth_day;
+    public BirthDayComparable(String birth_day) {
+        this.birth_day = birth_day;
+    }
+
+    @Override
+    public int compareTo(Employee o) {
+        if (o.getBirthDay().equals(this.birth_day))
+            return 0;
+        else
+            return -1;
+    }
+}
+
+public class Option2 {
+    public List<Employee> process_option(String option, List<Employee> employeeList, String column, String condition) {
+        if (column.equals("name")){
+            if(option.equals("-f")) {
+                Fitler<Employee> option_filter = new Fitler<Employee>();
+                FirstNameComparable fname = new FirstNameComparable(condition);
+
+                return option_filter.filter(employeeList, fname);
+            }
+
+            if(option.equals("-l")){
+                Fitler<Employee> option_filter = new Fitler<Employee>();
+                LastNameComparable lname = new LastNameComparable(condition);
+
+                return option_filter.filter(employeeList, lname);
+            }
+
+            throw new RuntimeException("Invalid name option("+ option +")");
```

Comment:
Invalid한 option을 파라미터로 전달받았거나 하는 경우 예외 처리를 하였는데, 이렇게 하는게 맞는 건지 잘 모르겠습니다.
이 부분은 아시는분은 조언 주시면 감사하겠습니다!ㅎㅎ

Reply:
Invalid한 option은 exception을 던지는 것으로 처리하는 것은 매우 좋은 것 같습니다!

추가적인 저의 개인 의견입니다:
1. `RuntimeException`보다 조금 더 descriptive한 `IllegalArgumentException`은 어떨까요?
2. Method signature에도 `throws RuntimeException` 등 을 붙여주면 `Option2` 클래스를 사용하는 consumer 입장에서 에러핸들링을 구현하도록 유도할 수 있을 것 같습니다!

Reply:
Invalid한 option은 exception을 던지는 것으로 처리하는 것은 매우 좋은 것 같습니다!

추가적인 저의 개인 의견입니다:
1. RuntimeException보다 조금 더 descriptive한 IllegalArgumentException은 어떨까요? https://github.com/mj610-kim/excellent-code/pull/18#issuecomment-1073378278 에 화정님도 같은 의견을 주셨네요!
2. Method signature에도 throws RuntimeException 등 을 붙여주면 Option2 클래스를 사용하는 consumer 입장에서 에러핸들링을 구현하도록 유도할 수 있을 것 같습니다!

Reply:
> 2. Method signature에도 `throws RuntimeException` 등 을 붙여주면 `Option2` 클래스를 사용하는 consumer 입장에서 에러핸들링을 구현하도록 유도할 수 있을 것 같습니다!

태민님 구두 질문으로 확인해본 결과, `RuntimeException`과 그의 subclass (`IllegalArgumentException` 포함)은 unchecked exception으로, method signature에 넣을 필요가 없어보입니다!

(나중에 잘못된 argument를 주는 것 자체를 막는 방법으로 parameter값을 `String`이 아닌 `enum` 등으로 바꾸는 작업들이 가능 할 것 같습니다.)

참고: https://stackoverflow.com/questions/20871763/should-i-declare-unchecked-exceptions-in-the-throws-specification#:~:text=Since%20unchecked%20exceptions%20indicate%20programming,highest%20level%20of%20your%20program.

Reply:
넵 감사합니다. 참고하여 수정하도록 하겠습니다 ㅎㅎ